packages/litexa/src/parser/state.coffee

```diff
@@ -300,7 +300,12 @@ class lib.State
       try
         model = intent.toModelV2(context)
       catch err
-        throw "failed to write language model for state `#{@name}`: #{err}"
+        if err.location
+          throw err
```

Comment:
It looks like `formatLocationStart` is used for formatting the location in an error message, so it's not used here. Digging into intent.toModelV2(), this is where the static conditional check can throw a ParserError, in which case we want to bubble that up instead of the more generic error. I've added a comment explaining this in the second revision.